libraries/oakestra_utils_library/oakestra_utils/types/statuses.py

```diff
@@ -0,0 +1,87 @@
+from typing import Optional
+
+from oakestra_utils.types.common import CustomEnum
+
+# Legend
+# - 🟨: Only mentioned in Schedulers.
+# - 🟦: Explicitly mentioned in the Go NodeEngine and in python code.
+# - 🟪: Only mentioned in the Go NodeEngine.
+# - All others are only explicitly mentioned in non-scheduler Python code.
+
+
+class Status(CustomEnum):
+    pass
+
+
+class LegacyStatus(Status):
+    """TODO: Investigate how these legacy statuses are used and how to replace them gracefully.
+
+    Note: These values had been ints before"""
+
+    LEGACY_0 = "0"
+    LEGACY_1 = "1"
+
+
+class SchedulingStatus(Status):
+    "Represents the state before a service gets deployed"
+    pass
+
+
+class NegativeSchedulingStatus(SchedulingStatus):
+    TARGET_CLUSTER_NOT_FOUND = "TargetClusterNotFound"  # 🟨
+    TARGET_CLUSTER_NOT_ACTIVE = "TargetClusterNotActive"  # 🟨
+    TARGET_CLUSTER_NO_CAPACITY = "TargetClusterNoCapacity"  # 🟨
+
+    NO_ACTIVE_CLUSTER_WITH_CAPACITY = "NoActiveClusterWithCapacity"  # 🟨
+
+    NO_WORKER_CAPACITY = "NO_WORKER_CAPACITY"
+    NO_QUALIFIED_WORKER_FOUND = "NO_QUALIFIED_WORKER_FOUND"  # 🟨
+    NO_NODE_FOUND = "NO_NODE_FOUND"  # 🟨
+
+
+class PositiveSchedulingStatus(SchedulingStatus):
+    REQUESTED = "REQUESTED"
+    CLUSTER_SCHEDULED = "CLUSTER_SCHEDULED"
+    NODE_SCHEDULED = "NODE_SCHEDULED"
+
+
+class DeploymentStatus(Status):
+    """Represents the state of a service during deployment,
+    after it got scheduled."""
+
+    CREATING = "CREATING"  # 🟪
+
+    RUNNING = "RUNNING"
+    # TODO: Figure out strong difference between ACTIVE and RUNNING states.
```

Comment:
In principle:

NODE_SCHEDULED: Container not yet created but node is working on it
ACTIVE:  Container created but task not running yet. Therefore the image was downloaded, container metadata created, but task process inside the container is not running yet. E.g., because we're doing other pre-flight operations such as attaching the network. We can use a more conventional name such as "CREATED", I would personally prefer that. 
RUNNING: Container created and task running

Reply:
Thank you for the clarification, I have adjusted the file accordingly.
I have renamed the ACTIVE status to CREATED.